src/main/native/com/code_intelligence/jazzer/android/native_agent.cpp

```diff
@@ -0,0 +1,323 @@
+
+    if ((bool)isModifiable) {
+      classesToRetransform.push_back(classes[i]);
+    }
+  }
+
+  jvmtiError errorNum = jvmti->RetransformClasses(classesToRetransform.size(),
+                                                  &classesToRetransform[0]);
+  if (errorNum != JVMTI_ERROR_NONE) {
+    std::cout << "Could not retransform classes. JVMTI error: " << errorNum
+              << std::endl;
+    exit(1);
+  }
+}
+
+std::vector<std::string> getDexFiles(std::string jarPath, JNIEnv* env) {
+  jclass jazzerClass = env->FindClass(kAndroidAgentClass.c_str());
+  if (jazzerClass == nullptr) {
+    std::cout << kAndroidAgentClass << " could not be found" << std::endl;
+    exit(1);
+  }
+
+  const char* getDexFilesFunction = "getDexFilesForJar";
+  jmethodID getDexFilesForJar =
+      env->GetStaticMethodID(jazzerClass, getDexFilesFunction,
+                             "(Ljava/lang/String;)[Ljava/lang/String;");
+  if (getDexFilesForJar == nullptr) {
+    std::cout << getDexFilesFunction << " could not be found\n";
+    exit(1);
+  }
+
+  jstring jJarFile = env->NewStringUTF(jarPath.data());
+  jobjectArray dexFilesArray = (jobjectArray)env->CallStaticObjectMethod(
+      jazzerClass, getDexFilesForJar, jJarFile);
+  int length = env->GetArrayLength(dexFilesArray);
+
+  std::vector<std::string> dexFilesResult;
+  for (int i = 0; i < length; i++) {
+    jstring dexFileJstring =
+        (jstring)env->GetObjectArrayElement(dexFilesArray, i);
+    const char* dexFileChars = env->GetStringUTFChars(dexFileJstring, NULL);
+    std::string dexFileString(dexFileChars);
+
+    env->ReleaseStringUTFChars(dexFileJstring, dexFileChars);
+    dexFilesResult.push_back(dexFileString);
+  }
+
+  return dexFilesResult;
+}
+
+void initializeBootclassOverrideJar(std::string jarPath, JNIEnv* env) {
+  std::vector<std::string> dexFiles = getDexFiles(jarPath, env);
+
+  std::cout << "Adding DEX files for: " << jarPath << std::endl;
+  for (int i = 0; i < dexFiles.size(); i++) {
+    std::cout << "DEX FILE: " << dexFiles[i] << std::endl;
+  }
+
+  for (int i = 0; i < dexFiles.size(); i++) {
+    std::cout << "Processing DEX File: " << dexFiles[i] << std::endl;
+    jclass bootHelperClass = env->FindClass(kAndroidAgentClass.c_str());
+    if (bootHelperClass == nullptr) {
+      std::cout << kAndroidAgentClass << " could not be found" << std::endl;
+      exit(1);
+    }
+
+    jmethodID getBytecodeFromDex =
+        env->GetStaticMethodID(bootHelperClass, "getBytecodeFromDex",
+                               "(Ljava/lang/String;Ljava/lang/String;J)[B");
+    if (getBytecodeFromDex == nullptr) {
+      std::cout << "'getBytecodeFromDex' not found\n";
+      exit(1);
+    }
+
+    jstring jjarPath = env->NewStringUTF(jarPath.data());
+    jstring jdexFile = env->NewStringUTF(dexFiles[i].data());
+    jlong joffset = (jint)0;
+
+    int length = 1;
+    std::vector<unsigned char> dexFileBytes;
+    while (length > 0) {
```

Comment:
It would seem to be simpler to have this loop in the `getBytecodeFromDex` Java method.